default.nix

```diff
@@ -62,13 +62,33 @@ let
       # the top-level build
       buildTopLevel =
         let
-          drv =
+          drv' =
             build
               {
                 pname = config.packageName;
                 inherit (config) userAttrs src;
                 builtDependencies = lib.optional (! config.isSingleStep) buildDeps;
               };
+
+          # If the project we're building uses CMake, let's proactively get rid
+          # of `CMakeCache.txt`.
+          #
+          # That's because this file contains a dump of environmental variables
+          # from the deps-only derivation and it could happen that the main
+          # derivation uses different env-vars - when this happens, CMake will
+          # fail, saying:
+          #
+          # ```
+          # CMake Error: The current CMakeCache.txt directory ... is different than the directory ... where CMakeCache.txt was created.
+          # ```
+          drv = drv'.overrideAttrs (attrs: attrs // {
+            preBuild = (attrs.preBuild or "") + ''
+              find \
+                  -name CMakeCache.txt \
+                  -exec rm {} \;
+            '';
+          });
```

Comment:
I've just checked it:
```nix
drv = drv'.overrideAttrs (attrs: attrs // {
  src = builtins.filterSource
    (name: type: (builtins.baseNameOf name) != "CMakeCache.txt")
    attrs.src;
});
```
... but it doesn't seem to work - most likely because the `CMakeCache.txt` comes not from `src` itself but rather from Naersk's `configurePhase`:

https://github.com/nix-community/naersk/blob/8507af04eb40c5520bd35d9ce6f9d2342cea5ad1/build.nix#L210

So perhaps a less confusing solution would be to put this `find ...` after that `for` loop, inside `configurePhase`? Something like this seems to work correctly:

```patch
diff --git a/build.nix b/build.nix
index cdfe73e..b324b9d 100644
--- a/build.nix
+++ b/build.nix
@@ -226,6 +226,11 @@ let
           fi
       done
 
+      find \
+          target \
+          -name CMakeCache.txt \
+          -exec rm {} \;
+
       export CARGO_HOME=''${CARGO_HOME:-$PWD/.cargo-home}
       mkdir -p $CARGO_HOME
 ```

Reply:
Actually, I think we could leave this up to the user, what do you think? And simply add an example/note of how to use naersk with CMake?

Reply:
Sure, this sounds good, too - lemme adjust the merge request, then 🙂

Reply:
alrighty, ready! :rocket: